src/backend/tools/web_scrape.py

```diff
@@ -0,0 +1,28 @@
+from typing import Any, Dict, List
+
+from bs4 import BeautifulSoup
+from requests import get
+
+from backend.tools.base import BaseTool
+
+
+class WebScrapeTool(BaseTool):
+    @classmethod
+    def is_available(cls) -> bool:
+        return True
+
+    def call(self, parameters: dict, **kwargs: Any) -> List[Dict[str, Any]]:
+        url = parameters.get("url")
+
+        page_content = get(url).text
```

Comment:
Check if the response was successful before fetching the text. Some pages are restricted, and a GET request may not be sufficient to retrieve the content

```suggestion
        response = get(URL)
        
        if not response.ok:
                return ["text": response.reason] # maybe something else?
         
         page_content = response.text        
```

Reply:
great callout. thanks